data/flow/flowdata.py

```diff
@@ -180,14 +190,31 @@ def _parse_function(example_proto):
             image_b = tf.reshape(image_b, [H, W, 3])
             flow = tf.decode_raw(parsed_features[FLOW_RAW], tf.float32)
             flow = tf.reshape(flow, [H, W, 2])
+
+            # Cropping augmentation.
+            image_a, image_b, flow = tf_random_crop([image_a, image_b, flow], self.crop_size)
+
+            if do_augmentations:
+                config = {
+                    'contrast_min': 0.8, 'contrast_max': 1.3,
+                    'gamma_min': 0.7, 'gamma_max': 1.5,
+                    'gain_min': 0.8, 'gain_max': 1.25,
+                    'brightness_stddev': 0.2,
+                    'hue_min': -0.3, 'hue_max': 0.3,
+                    'noise_stddev': 0.04
+                }
+                image_a, image_b = tf_image_augmentation([image_a, image_b], config)
+
             return image_a, image_b, flow
 
         dataset = tf.data.TFRecordDataset(filenames)
-        dataset = dataset.map(_parse_function)
-        dataset = dataset.shuffle(buffer_size=250)
-        dataset = dataset.batch(self.batch_size)
         if repeat:
-            dataset = dataset.repeat()
+            dataset = dataset.apply(tf.contrib.data.shuffle_and_repeat(buffer_size=len(filenames)))
+        else:
+            dataset = dataset.shuffle(buffer_size=len(filenames))
+        dataset = dataset.map(_parse_function, num_parallel_calls=multiprocessing.cpu_count())
+        dataset = dataset.batch(self.batch_size)
+        dataset = dataset.prefetch(self.batch_size)
```

Comment:
At this point each element in the dataset is a batch, did you mean to prefetch with buffer_size=1 instead?

Reply:
That's probably true,